nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -602,29 +605,31 @@ nav_msgs::msg::Path RegulatedPurePursuitController::transformGlobalPlan(
   const double max_costmap_dim = std::max(costmap->getSizeInCellsX(), costmap->getSizeInCellsY());
   const double max_costmap_extent = max_costmap_dim * costmap->getResolution() / 2.0;
 
-  // Find the first pose that turns in the opposite direction from the first pose
-  findPathUpToTurn(global_plan_, M_PI);
-
-  const double max_transform_dist = std::min(max_costmap_extent, dist_to_cusp);
-
+  // This bound is based on the assumption that the robot can't have traveled very far
+  // since the last iteration, and that when the global_plan was first passed, the robot
+  // was intended to start at the beginning.
+  // lookahead distance should be a pretty good proxy for this bound.
   auto closest_pose_upper_bound =
     nav2_util::geometry_utils::first_element_beyond(
-    global_plan_.poses.begin(), global_plan_.poses.end(), max_transform_dist);
+    global_plan_.poses.begin(), global_plan_.poses.end(), lookahead_distance);
```

Comment:
Revert this back to `max_transform_dist` and have it be computed to be either distance to direction change **or** max parameterization rather than the costmap size.

Reply:
We should not be pruning the total path available to the algorithm by the lookahead distance, that's way too short

Reply:
Keep in mind this is not the pruning distance: this is the upper bound for finding the beginning of the transformed path. So this upper bound should be relatively short.

Reply:
I don't think the lookahead distance though is big enough. This works fine if you're talking about a robot with a large look ahead distance, but if its short & the controller is running slow (10hz or something) due to limited compute, I could see cycles where that would find invalid past points because you hit the bound.

I think the role of this should be the largest bound reasonably possible. That's kind of why I would support this being a parameterization

Reply:
For both locations where `max_transform_dist` is used (representing currently min of costmap size and directional change) could be replaced with the same parameter where instead of costmap size, now its a parameterized size. I think then using the updated `max_transform_dist` would work for both cases: (1) pruning the path forward and (2) finding the max to search for the closest point on the robot. 

What do you think?